### DISCUSSION TITLE
COBRA and RAVEN toolboxes

### DISCUSSION THREAD

**Discussion by oykunazduranlar** [SysBioChalmers/RAVEN#517] [General]
Hello! First of all, I'm not very familiar with the COBRA and RAVEN toolboxes. I have a few questions to better understand. Firstly, what are the differences between these toolboxes, and in which situations would it be more appropriate to use one over the other? Secondly, can we use both toolboxes together while working in MATLAB?

**Comment by edkerk** [comment]
Having to contrast COBRA and RAVEN, then COBRA's strength is that it has more simulation algorithms implemented, to be applied on existing models.

RAVEN's strength is then the construction of new models from homology or KEGG, curation of models, making context/tissue-specific models with tINIT, etc. RAVEN also has simulation functionality (FBA, FVA, gene essentiality, FSEOF, etc.), but less than COBRA.

Both toolboxes can be used together while working in MATLAB: all their functions have unique names. The only thing to pay attention is that the model structure is somewhat different (see also#184, mostly related to how compartments are handled and how reaction/gene/metabolite annotation to external databases (e.g. MetaCyc) are provided). For this, `ravenCobraWrapper` can convert a RAVEN-style model to COBRA-style model and vice versa.
